src/main/kotlin/com/demonwav/mcdev/translations/TranslationFiles.kt

```diff
@@ -0,0 +1,299 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2018 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.translations
+
+import com.demonwav.mcdev.facet.MinecraftFacet
+import com.demonwav.mcdev.platform.mcp.McpModuleType
+import com.demonwav.mcdev.translations.index.TranslationIndex
+import com.demonwav.mcdev.translations.index.TranslationInverseIndex
+import com.demonwav.mcdev.translations.lang.LangFile
+import com.demonwav.mcdev.translations.lang.LangFileType
+import com.demonwav.mcdev.translations.lang.gen.psi.LangEntry
+import com.demonwav.mcdev.translations.lang.gen.psi.LangTypes
+import com.demonwav.mcdev.translations.sorting.EmptyLine
+import com.demonwav.mcdev.translations.sorting.Key
+import com.demonwav.mcdev.translations.sorting.Template
+import com.demonwav.mcdev.translations.sorting.TemplateElement
+import com.demonwav.mcdev.util.SemanticVersion
+import com.demonwav.mcdev.util.applyWriteAction
+import com.demonwav.mcdev.util.mcDomain
+import com.demonwav.mcdev.util.mcPath
+import com.intellij.ide.DataManager
+import com.intellij.json.JsonElementTypes
+import com.intellij.json.JsonFileType
+import com.intellij.json.psi.JsonObject
+import com.intellij.json.psi.JsonProperty
+import com.intellij.json.psi.JsonStringLiteral
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.PsiNamedElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.util.DocumentUtil
+import com.intellij.util.indexing.FileBasedIndex
+import java.util.Locale
+
+object TranslationFiles {
+    private val MC_1_12_2 = SemanticVersion.release(1, 12, 2)
+
+    fun isTranslationFile(file: VirtualFile) =
+        file.mcPath?.startsWith("lang/") == true && file.fileType in listOf(LangFileType, JsonFileType.INSTANCE)
```

Comment:
Why did you add `== true` explicitly here?

Reply:
Since `mcPath` is nullable, you need to check for an actual value.